Messaging/Plugin.Messaging.Android/SmsTask.cs

```diff
@@ -38,16 +61,57 @@ public void SendSms(string recipient = null, string message = null)
 
         public void SendSmsInBackground(string recipient, string message = null)
         {
-            message = message ?? string.Empty;
+            if (!CanSendSmsInBackground)
+            {
+                OnSmsDeliveryResult?.Invoke(isSuccessful: false);
+
+                return;
+            }
+
+            message ??= string.Empty;
+
+            _smsManager = SmsManager.Default;
 
-            if (CanSendSmsInBackground)
+            var piSent = PendingIntent.GetBroadcast(CONTEXT, 0, new Intent(SMS_SENT_INTENT_ACTION), 0);
+            var piDelivered = PendingIntent.GetBroadcast(CONTEXT, 0, new Intent(SMS_DELIVERY_INTENT_ACTION), 0);
+
+            CONTEXT.RegisterReceiver(this, new IntentFilter(SMS_SENT_INTENT_ACTION));
+            CONTEXT.RegisterReceiver(this, new IntentFilter(SMS_DELIVERY_INTENT_ACTION));
+
+            _smsManager.SendTextMessage(recipient, null, message, piSent, piDelivered);
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private void OnReceiveAction(bool result)
+        {
+            UnRegisterBroadcastReceiver();
+
+            OnSmsDeliveryResult?.Invoke(isSuccessful:result);
+        }
+
+        private void UnRegisterBroadcastReceiver()
+        {
+            try
+            {
+                CONTEXT.UnregisterReceiver(this);
+            }
+            catch (Java.Lang.IllegalArgumentException)
             {
-                var smsManager = SmsManager.Default;
-                var msgs = smsManager.DivideMessage(message);
-                smsManager.SendMultipartTextMessage(recipient, null, msgs, null, null);
             }
         }
 
         #endregion
+
+        #region Private Field
+
+        private const string SMS_DELIVERY_INTENT_ACTION = "BSN.Resa.DoctorApp.SMS_Delivery";
```

Comment:
Why this specific project dependent string declared here? please remove it. or if you need it, please generals it